From e804c6a3b4626a3a0a8f3000cbf3ed887116f911 Mon Sep 17 00:00:00 2001
From: ghoshp <purnendu.ghosh@intel.com>
Date: Tue, 21 Aug 2018 12:05:15 +0530
Subject: [PATCH] New QoS Mgr extended hook is added. The new extended hook is
 required if TC operation has to be done on the netdevice which are not
 registered to DP. This is extended to take dequeue index and port_id.

---
 drivers/net/ethernet/lantiq/ppa/ppa_hook.c | 3 +++
 include/net/ppa/qos_mgr_tc_hook.h          | 1 +
 2 files changed, 4 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/ppa/ppa_hook.c b/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
index 1cabd20225ae..0e563e418238 100644
--- a/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
+++ b/drivers/net/ethernet/lantiq/ppa/ppa_hook.c
@@ -417,6 +417,9 @@ int32_t (*ppa_hook_reset_qos_wfq)(uint32_t portid, uint32_t queueid, uint32_t fl
 int32_t (*qos_mgr_hook_setup_tc)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc) = NULL;
 EXPORT_SYMBOL(qos_mgr_hook_setup_tc);
 
+int32_t (*qos_mgr_hook_setup_tc_ext)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc, int32_t deq_idx, int32_t port_id) = NULL;
+EXPORT_SYMBOL(qos_mgr_hook_setup_tc_ext);
+
 #ifdef CONFIG_INTEL_IPQOS_MARK_SKBPRIO
 /*
  * Function to mark priority based on specific criteria
diff --git a/include/net/ppa/qos_mgr_tc_hook.h b/include/net/ppa/qos_mgr_tc_hook.h
index d748bbc1e077..98c07c083469 100644
--- a/include/net/ppa/qos_mgr_tc_hook.h
+++ b/include/net/ppa/qos_mgr_tc_hook.h
@@ -25,6 +25,7 @@
  */
 #ifdef __KERNEL__
 extern int32_t (*qos_mgr_hook_setup_tc)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc);
+extern int32_t (*qos_mgr_hook_setup_tc_ext)(struct net_device *dev, u32 handle, __be16 protocol, struct tc_to_netdev *tc, int32_t deq_idx, int32_t port_id);
 #endif /*end of __KERNEL__*/
 /* @} */
 #endif  /*  __QOS_MGR_TC_HOOK_H */
